marlowe-dashboard-client/src/Pickup/View.purs

```diff
@@ -132,15 +129,15 @@ pickupNewWalletCard wallets newWalletDetails pickupNewWalletContext =
               [ text "Cancel" ]
           , button
               [ classNames $ Css.primaryButton <> [ "flex-1" ]
-              , disabled $ isJust mWalletNicknameError || isJust mContractInstanceIdError
+              , disabled $ isJust mWalletNicknameError || isJust mContractInstanceIdError || pickingUp
               , onClick_ PickupNewWallet
               ]
-              [ text "Pickup" ]
+              [ text if pickingUp then "Picking up... " else "Pickup" ]
```

Comment:
Besides the informative text, there are some designs Russell provided for loading stuff, which are a perfect match for this already available tailwind animation https://tailwindcss.com/docs/animation#pulse